pkg/kubelet/prober/results/results_manager.go

```diff
@@ -40,14 +40,17 @@ type Manager interface {
 }
 
 // Result is the type for probe results.
-type Result bool
+type Result int
 
 const (
-	// Success is encoded as "true" (type Result)
-	Success Result = true
+	// Success is encoded as 0 (type Result)
+	Success Result = 0
 
-	// Failure is encoded as "false" (type Result)
-	Failure Result = false
+	// Failure is encoded as 1 (type Result)
+	Failure Result = 1
+
+	// Unknown is encoded as -1 (type Result)
+	Unknown Result = -1
```

Comment:
@odinuge is that what you meant? Of course I have no way of  testing that from work :-(

Reply:
Yes! So as long as this doesn't have any unintended side effects anywhere else (the constants are exported), this looks good to me, but ill defer that to someone with more knowledge about the code than me.  There are at least some tests that assume `Result` is a bool.

After merging https://github.com/kubernetes/kubernetes/pull/84179 and https://github.com/kubernetes/kubernetes/pull/84291 into this branch, all the e2e tests pass! :tada: 

The tests also pass when reverting the new code block below `case updateStartup := <-kl.startupManager.Updates():` in `pkg/kubelet/kubelet.go`, so I think that can be removed.

The new test suite is also merged and can be found here: https://testgrid.k8s.io/sig-node-kubelet#node-kubelet-serial-alpha. It does however suffer the same issue as the other node-e2e test suits atm., and that is being tracked here: https://github.com/kubernetes/kubernetes/issues/84236

Reply:
I am cleaning up the unit tests...
Regarding #84179 and #84291, should I put everything in this branch?

Reply:
I think we can keep them as they are, in order to make reviewing a bit simpler.

Reply:
Ill defer the naming of this to the approvers (eg. `Pending`)

Reply:
One small concern I have is I am not sure if anyone already depend on the previous defined results. cc/ @yujuhong